### PR TITLE
feat(List): add CellRenderer for showing text + icon

### PR DESCRIPTION
**Pull request by vyshkov**
**What is the problem this PR is trying to solve?**

We need to show icon right after the text in table cell like on the picture:

https://jira.talendforge.org/secure/attachment/132019/132019_screenshot-1.png

**What is the chosen solution to this problem?**

Custom Cell renderer

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

### CODE REVIEW

**Comment by vyshkov** [comment]
http://940.talend.surge.sh/components/?selectedKind=List&selectedStory=Table%20icons&full=0&down=1&left=1&panelRight=0&downPanel=storybook%2Factions%2Factions-panel

File: packages/components/src/VirtualizedList/CellTextIcon/CellWithIcon.component.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+CellWithIcon.displayName = 'VirtualizedList(CellWithIcon)';
+CellWithIcon.propTypes = {
+	// The cell value : props.rowData[props.dataKey]
+	cellData: PropTypes.string,
+	// Can be any object
+	rowData: PropTypes.object,
+	columnData: PropTypes.shape({
+		getIcon: PropTypes.func,
```
Comment (acateland) [inline packages/components/src/VirtualizedList/CellTextIcon/CellWithIcon.component.js]: Grticon should be tagged as required

File: packages/components/src/VirtualizedList/CellTextIcon/CellWithIcon.component.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import Action from '../../Actions/Action';
+import styles from './CellWithIcon.scss';
+
+/**
+ * Cell renderer that displays a badge
+ */
+function CellWithIcon(props) {
```
Comment (acateland) [inline packages/components/src/VirtualizedList/CellTextIcon/CellWithIcon.component.js]: Please spread value in the function declaration so the alpi is more clear without diging into the code

File: packages/components/src/VirtualizedList/CellTextIcon/CellWithIcon.component.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	const action = props.columnData.getIcon(props.rowData);
+
+	return (
+		<div className={styles['cell-icon-container']}>
```
Comment (yyanwang) [inline packages/components/src/VirtualizedList/CellTextIcon/CellWithIcon.component.js]: please use something like `classNames('cell-icon-container', styles['cell-icon-container'])` so style can be customized elsewhere.

**Pull request merged**